include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -58,13 +58,14 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __f
         oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _RandomAccessIterator2>();
     auto __buf2 = __keep2(__first2, __first2 + __n);
 
-    _RepackedTp __res = oneapi::dpl::__par_backend_hetero::__parallel_transform_reduce<_RepackedTp>(
-        ::std::forward<_ExecutionPolicy>(__exec),
-        unseq_backend::transform_init<_Policy, _BinaryOperation1, _Functor>{__binary_op1,
-                                                                            _Functor{__binary_op2}}, // transform
-        __binary_op1,                                                                                // combine
-        unseq_backend::reduce<_Policy, _BinaryOperation1, _RepackedTp>{__binary_op1},                // reduce
-        __buf1.all_view(), __buf2.all_view());
+    auto __res = oneapi::dpl::__par_backend_hetero::__parallel_transform_reduce<_RepackedTp>(
+                     ::std::forward<_ExecutionPolicy>(__exec),
+                     unseq_backend::transform_init<_Policy, _BinaryOperation1, _Functor>{
+                         __binary_op1, _Functor{__binary_op2}},                                    // transform
+                     __binary_op1,                                                                 // combine
+                     unseq_backend::reduce<_Policy, _BinaryOperation1, _RepackedTp>{__binary_op1}, // reduce
+                     __buf1.all_view(), __buf2.all_view())
+                     .get();
 
     return __binary_op1(__init, _Tp{__res});
```

Comment:
There is a host call of `__binary_op1`.  I have a couple of questions:
1) @capatober, Why that computation is not executed on a device?
2) @reble , what are you going to do with it from an async call perspective?

Reply:
Here we work with host arguments: `__init` and `__res` are both on hosts.
@reble, am I right that this issue is resolved at async parallel_reduce pattern?

Reply:
`__first1` also my be on the host... but we pass dpc++ policy and we expect computation on a device :-)
Probably,   `__init `  should be copied to a device as well? And what if  `__init ` is on a device already? (as a result one of previous operations)